### PR TITLE
feat(cat-voices): wallet link flow structure

### PR DESCRIPTION
**Pull request by dtscalac**
# Description

Adds a structure for link wallet dialog flow.

## Related Issue(s)

Refers #818

## Screenshots

Screenshot 2024-09-24 at 21 19 54

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/lib/widgets/buttons/specialized/voices_learn_more_button.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+/// A "Learn More" button that redirects usually to an external content.
+class VoicesLearnMoreButton extends StatelessWidget {
```
Comment (damian-molinski) [inline catalyst_voices/lib/widgets/buttons/specialized/voices_learn_more_button.dart:7]: Is `voices_buttons.dart` better place for such button?

eg. https://github.com/input-output-hk/catalyst-voices/pull/881/files#diff-a9d554894c1a22e2e2e04448cab4a67fd40a1c6fbd490569cac61cb009f28784R152

File: catalyst_voices/lib/pages/registration/link_wallet/link_wallet_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import 'package:catalyst_voices/pages/registration/link_wallet/intro/link_wallet_intro_dialog.dart';
+import 'package:catalyst_voices/pages/registration/link_wallet/link_wallet_stage.dart';
+import 'package:catalyst_voices/pages/registration/link_wallet/select_wallet/select_wallet_dialog.dart';
+import 'package:catalyst_voices/widgets/modals/voices_dialog.dart';
+import 'package:flutter/material.dart';
+
+/// The link wallet flow consisting
+/// of [LinkWalletStage]'s during the registration.
+class LinkWalletDialog extends StatefulWidget {
+  const LinkWalletDialog({super.key});
```
Comment (damian-molinski) [inline catalyst_voices/lib/pages/registration/link_wallet/link_wallet_dialog.dart:11]: this constructor can be private so only `show` is access point here

File: catalyst_voices/lib/widgets/buttons/specialized/voices_learn_more_button.dart (hunk 1 above)
Comment (dtscalac) [reply catalyst_voices/lib/widgets/buttons/specialized/voices_learn_more_button.dart:7]: Agree, moved there

File: catalyst_voices/lib/pages/registration/link_wallet/link_wallet_dialog.dart (hunk 2 above)
Comment (dtscalac) [reply catalyst_voices/lib/pages/registration/link_wallet/link_wallet_dialog.dart]: Fixed

**Review by damian-molinski — APPROVED** [review]
lgtm

**Pull request merged**